### DISCUSSION TITLE
Trouble logging in to the Developer Portal with email

### DISCUSSION THREAD

**Discussion by pete0emerson** [gruntwork-io/knowledge-base#395] [Help] [answered]
When I click the “Sign In” link it says the link is expired. I did it from the button and from copy/pasting the actual URL into the browser.

What are next steps?

---

<ins datetime="2022-04-27T17:13:54Z">
  <p><a href="https://gruntwork.zendesk.com/agent/tickets/108516">Tracked in ticket #108516</a></p>
</ins>

**Comment by pete0emerson** [comment, marked as answer]
Some email providers  will pre-visit the link when you click on it, which can invalidate the token.

To remedy this:

1. Request a new login token
2. When you get the email, don't click on the "Sign In" link (this would invalidate the token prematurely).
3. Copy the URL below the "Trouble Signing In?" header, and paste it into your browser.
4. This should log you in.

**Reply by josh-padnick** [reply]
To add some additional detail here, there are some tools like [Defender for Office 365](https://docs.microsoft.com/en-us/microsoft-365/security/office-365-security/safe-links?view=o365-worldwide) that automatically visit all the links in your email to see if they're malicious. Unfortunately, this breaks our portal's sign-in model of "you can only click this button once before the link expires." That's why we include the separate link you can copy and paste, which tools like Defender often don't automatically visit.
